### ISSUE TITLE
OPTIONS request does not always send x-user-token / x-user-path

### ISSUE THREAD

**Issue by pallix** [liqd/adhocracy3.mercator#1557]
When logged in as a moderator, and trying to access a document created by god, the frontend send an OPTION request on this URL:

http://localhost:6551/r/alexanderplatz/workshopverfahren-2015/document_0000000/

The x-user-token and x-user-path values are not send.

**Comment by xi** [comment]
This is most likely a [preflight request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests). That is send by the browser automatically and has nothing to do with the frontend. The frontend should send a separate OPTIONS request that has the right headers set.

**Comment by pallix** [comment]
One of the call originated from angular.js. Can it still be a preflight request?

**Comment by xi** [comment]
> One of the call originated from angular.js. Can it still be a preflight request?

Can you tell me where exactly this was triggered?

**Comment by pallix** [comment]
For instance when accessing `http://localhost:6551/r/organisation/alexanderplatz/` as moderator, there is an OPTIONS request on `http://localhost:6541/organisation/` with the following headers:

```
Accept:*/*
Accept-Encoding:gzip, deflate, sdch
Accept-Language:en-GB,en-US;q=0.8,en;q=0.6
Access-Control-Request-Headers:accept, x-user-path, x-user-token
Access-Control-Request-Method:GET
Cache-Control:no-cache
Connection:keep-alive
Host:localhost:6541
Origin:http://localhost:6551
Pragma:no-cache
Referer:http://localhost:6551/r/organisation/alexanderplatz/
User-Agent:Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.155 Safari/537.36
```

The request is issued by Angular and the stacktrace is:

```
(anonymous function)	@	angular.js:10514
sendReq	@	angular.js:10333
$get.serverRequest	@	angular.js:10045
processQueue	@	angular.js:14567
(anonymous function)	@	angular.js:14583
$get.Scope.$eval	@	angular.js:15846
$get.Scope.$digest	@	angular.js:15657
$get.Scope.$apply	@	angular.js:15951
done	@	angular.js:10364
completeRequest	@	angular.js:10536
requestLoaded	@	angular.js:10477
```

**Comment by xi** [comment]
![snapshot-2015-08-19](https://cloud.githubusercontent.com/assets/202576/9328669/f8bb79be-45aa-11e5-95a1-94912fcc0da1.png)

I checked again: I have 5 OPTIONS and 3 GET requests. All request with Initiator *Other* do have x-user-token and x-user-path headers (including the OPTIONS request). All requests with Initiator *angular.js* (all OPTIONS) do not have these headers. So I guess these are the preflight requests.

This is a bit confusing in the chromium developer tools, but otherwise works as expected.

I also checked the frontend code again. I do not see how the frontend could ever skip the authentication headers on an options request. `adhHttp.options()` makes sure that the headers are registered before sending a request.
